### PR TITLE
Add implicit reference to `Microsoft.AspNetCore.App.Internal.Assets`

### PR DESCRIPTION
**Pull request by MackinnonBuck**
Follow up to https://github.com/dotnet/aspnetcore/pull/58721

### CODE REVIEW

**Comment by MackinnonBuck** [comment]
Tested locally by building a local version of the SDK, manually making changes to the generated `Microsoft.NETCoreSdk.BundledVersions.props` file, copying the SDK to a locally-built version of ASP.NET Core, undoing the workaround [here](https://github.com/dotnet/aspnetcore/blob/8d0f798cc4de54a2851748be635a58eadbf79593/src/ProjectTemplates/TestInfrastructure/Directory.Build.props.in#L22-L33), and running a project generated by the Blazor Web App project template. Everything seemed to work (the fingerprinted `blazor.web.js` file was loaded and the binlog showed that the implicit project reference was being added), so I'm going to mark this PR as ready-for-review.

File: src/Installer/redist-installer/targets/GenerateBundledVersions.targets [hunk 1]
```diff
@@ -590,6 +590,10 @@ Copyright (c) .NET Foundation. All rights reserved.
                              TargetFramework="net10.0"
                              WebAssemblySdkPackVersion="$(MicrosoftNETCoreAppRuntimePackageVersion)" />
 
+    <KnownAspNetCorePack Include="Microsoft.AspNetCore.App.Internal.Assets"
+                             TargetFramework="net10.0"
+                             AspNetCorePackVersion="$(MicrosoftAspNetCoreAppRuntimePackageVersion)" />
+
```
Comment (javiercn) [inline src/Installer/redist-installer/targets/GenerateBundledVersions.targets:596]: This is the only part I'm not 100% sure about. You can have an SDK and multiple runtimes. Meaning that you can have an  install that has 1.0.0, 1.0.1, and 1.0.2 installed. 1.0.2 might be bundled, but your app might be using `<FrameworkReference="Microsoft.AspNetCore.App" Version="1.0.0" />` or have it pinned in the `global.json`. 

In that case shouldn't we respect that?

@maraf do you have any thoughts?

Comment (javiercn) [reply src/Installer/redist-installer/targets/GenerateBundledVersions.targets:596]: Similar question. Do we have a way for this to only apply to 10+?

Comment (javiercn) [reply src/Installer/redist-installer/targets/GenerateBundledVersions.targets:596]: @maraf but it depends on whether you are pinning explicitly to a previous runtime, isn't it?

**Review by javiercn — APPROVED** [review]
Let's go with this for now

**Pull request merged**